src/go/pt-k8s-pxc-recovery/recover/recover.go

```diff
@@ -0,0 +1,297 @@
+package recover
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/percona/percona-toolkit/src/go/pt-k8s-pxc-recovery/kubectl"
+)
+
+type Cluster struct {
+	Name          string
+	Size          int
+	MostRecentPod int
+	Namespace     string
+	ClusterImage  string
+}
+
+func (c *Cluster) SetClusterSize() error {
+	args := []string{
+		"get",
+		"pxc",
+		c.Name,
+		"-o",
+		"jsonpath='{.spec.pxc.size}'",
+	}
+	strSize, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return err
+	}
+	strSize = strings.Trim(strSize, "'")
+	c.Size, err = strconv.Atoi(strSize)
+	if err != nil {
+		return fmt.Errorf("error getting cluster size, %s", err.Error())
+	}
+	return nil
+}
+
+func (c *Cluster) GetClusterImage() error {
+	args := []string{
+		"get",
+		"pod",
+		c.Name + "-pxc-0",
+		"-o",
+		"jsonpath='{.spec.containers[0].image}'",
+	}
+	clusterImage, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return err
+	}
+	c.ClusterImage = strings.Trim(clusterImage, "'")
+	return nil
+}
+
+func (c *Cluster) getPods() ([]string, error) {
+	args := []string{
+		"get",
+		"pods",
+	}
+	out, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return []string{}, err
+	}
+	formatedOutput := strings.Split(out, "\n")
+	podNames := []string{}
+	for _, v := range formatedOutput {
+		podName := strings.Split(v, " ")[0]
+		if strings.Contains(podName, c.Name) && strings.Contains(podName, "pxc") {
+			podNames = append(podNames, podName)
+		}
+	}
+	return podNames, nil
+}
+
+func (c *Cluster) ConfirmCrashedStatus() error {
+	podNames, err := c.getPods()
+	if err != nil {
+		return err
+	}
+	failedNodes := 0
+	for _, pod := range podNames {
+		logs, err := kubectl.RunCmd(c.Namespace, "logs", pod)
+		if err != nil {
+			return fmt.Errorf("error confirming crashed cluster status %s", err.Error())
+		}
+		if strings.Contains(logs, "grastate.dat") && strings.Contains(logs, "safe_to_bootstrap") {
+			failedNodes++
+		}
+	}
+	if len(podNames) != failedNodes {
+		return fmt.Errorf("found more than one pod running, can't use recovery tool, please restart failed pods manually")
+	}
+	return nil
+}
```

Comment:
What this code intends to do? _I see what it does but I'm interested in what **should** be done_
The predicate `len(podNames) != failedNodes`  and error "found more than one pod running..." are confusing. Why "more than one"? Is it ok to have one pod not crashed and running? If so - this code doesn't allow this

Reply:
it is supposed to make sure that all pods that are up are in crashed status and if not the recovery tool is not needed you can just restart the failed pods and it will re join the cluster